### PR TITLE
#7Task. ivan.lozovskij

### PR DESCRIPTION
**Pull request by Lozovskij**
https://travis-ci.org/github/Lozovskij/js-assignments/builds/706239499

### CODE REVIEW

File: task/07-yield-tasks.js [hunk 1]
```diff
@@ -108,9 +137,24 @@ function* depthTraversalTree(root) {
...
+    var ind = 0;
+    while(stack.length > ind) {
+        addChildren(stack, ind);
+        ind++;
+    }
+    for (let i = 0; i < stack.length; i++) {
+        yield stack[i];
+    }
+}
+function addChildren(stack, ind) {
```
Comment (elizavetaShcherbatykhAis) [inline task/07-yield-tasks.js]: You can't create function outside the main function, sorry

File: task/07-yield-tasks.js [hunk 2]
```diff
@@ -145,7 +200,10 @@ function* mergeSortedSequences(source1, source2) {
  *   Most popular implementation of the logic in npm https://www.npmjs.com/package/co
  */
 function async(generator) {
-    throw new Error('Not implemented');
+    var gen = generator();
+    var start = gen.next();//start.value is a Promise
+    var toTheEnd = (item) => item.done ? item.value : item.value.then((handover) => toTheEnd(gen.next(handover)));
+    return toTheEnd(start);//asynchronously gets the last value of the generator sequence
```
Comment (elizavetaShcherbatykhAis) [inline task/07-yield-tasks.js:207]: Thank you for the comments!

**Comment by elizavetaShcherbatykhAis** [comment]
Please check the comments.
I'll mark the PR as draft please click "ready for review" when it will be corrected!
Thank you!

**Comment by Lozovskij** [comment]
https://travis-ci.org/github/Lozovskij/js-assignments/builds/710143632

**Comment by elizavetaShcherbatykhAis** [comment]
Well done! I close PR as accepted

**Pull request closed without merging**